buildflow/core/app/runtime/actors/runtime.py

```diff
@@ -159,55 +163,71 @@ def is_active(self):
 
     async def _runtime_autoscale_loop(self):
         logging.info("Runtime checkin loop started...")
-        while self._status == RuntimeStatus.RUNNING:
-            for processor_pool in self._processor_pool_refs:
-                if self._status != RuntimeStatus.RUNNING:
-                    break
-
-                try:
-                    processor_snapshot: ProcessorSnapshot = (
-                        await processor_pool.actor_handle.snapshot.remote()
-                    )
-                except RayActorError:
-                    logging.exception("process actor unexpectedly died. will restart.")
-                    options = self.options.processor_options[
-                        processor_pool.processor.processor_id
-                    ]
-                    processor_pool.actor_handle = (
-                        PipelineProcessorReplicaPoolActor.remote(
-                            self.run_id,
-                            processor_pool.processor,
-                            options,
+        last_autoscale_event = datetime.utcnow()
+        # We keep running the loop while the job is running or draining to ensure
+        # we don't exit the main process before the drain is complete.
+        while (
```

Comment:
@JoshTanke just wanted a sanity check here. I think we also want this to run while it's draining otherwise this loop may exit before a drain is complete causing the main program to exit. Does that sound right?

Reply:
yeah, but Im wondering if we should separate this into 2 async loops. Maybe just add a TODO comment to explore that idea and give a little context on the 2 tasks this loop is doing

Reply:
added a TODO